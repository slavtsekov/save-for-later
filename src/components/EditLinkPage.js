import React from 'react';

import LinkForm from './LinkForm';

class EditLinkPage extends React.Component {
    link = {
        title: 'New Link',
        url: 'link.com',
        description: 'New Link',
        folder: 'folder1'
    };

    onSubmit = (data) => {
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Edit Link</h1>
                <LinkForm link={this.link} onSubmit={this.onSubmit} />
            </div>
        );
    }
}

export { EditLinkPage as default };
