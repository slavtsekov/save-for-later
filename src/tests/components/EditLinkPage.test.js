import React from 'react';
import { shallow } from 'enzyme';
import EditLinkPage from '../../components/EditLinkPage';

test('should render DashboardPage', () => {
    const wrapper = shallow(<EditLinkPage />);
    expect(wrapper).toMatchSnapshot();
});


test('should redirect to home page on submit', () => {
    const historyMock = { push: jest.fn() };
    const wrapper = shallow(<EditLinkPage history={historyMock} />);

    wrapper.find('LinkForm').prop('onSubmit')();

    expect(historyMock.push).toHaveBeenLastCalledWith('/');
});
