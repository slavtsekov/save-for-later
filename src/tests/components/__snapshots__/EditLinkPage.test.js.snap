// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage 1`] = `
<div>
  <h1>
    Add Link
  </h1>
  <LinkForm
    link={
      Object {
        "description": "New Link",
        "folder": "folder1",
        "title": "New Link",
        "url": "link.com",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
